stdlib/public/runtime/HaikuPort.cpp

```diff
@@ -0,0 +1,207 @@
+//===--- HaikuPort.cpp - Haiku ELF image inspection --------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+///
+/// This file includes the Haiku specific implementation of the Image Inspection
+/// routines specified in ImageInspectionELF. Since Haiku does not use 
+/// dl_iterate_phdr(), a local implementation is used here instead.
+/// 
+///
+//===----------------------------------------------------------------------===//
+
+#if defined(__HAIKU__)
+
+#include "ImageInspection.h"
+#include "swift/Runtime/Debug.h"
+#include <dlfcn.h>
+#include <elf.h>
+#include <string.h>
+#include <unistd.h>
+#include <OS.h>
+#include <image.h>
+
+struct dl_phdr_info {
+    void *dlpi_addr;
+    const char *dlpi_name;
+};
+
+#define RTLD_NOLOAD RTLD_LOCAL
+
+using namespace swift;
+
+/// The symbol name in the image that identifies the beginning of the
+/// protocol conformances table.
+static const char ProtocolConformancesSymbol[] =
+  ".swift2_protocol_conformances_start";
+/// The symbol name in the image that identifies the beginning of the
+/// type metadata record table.
+static const char TypeMetadataRecordsSymbol[] =
+  ".swift2_type_metadata_start";
+
+/// Context arguments passed down from dl_iterate_phdr to its callback.
+struct InspectArgs {
+  /// Symbol name to look up.
+  const char *symbolName;
+  /// Callback function to invoke with the metadata block.
+  void (*addBlock)(const void *start, uintptr_t size);
+  /// Set to true when initialize*Lookup() is called.
+  bool didInitializeLookup;
+};
+
+// Haiku implementation of dl_iterate_phdr
+static int dl_iterate_phdr(int (*callback)(struct dl_phdr_info *info, size_t size, void *data), void *data);
+
+static int dl_iterate_phdr(int (*callback)(struct dl_phdr_info *info, size_t size, void *data), void *data) {
+    pid_t team_id = getpid();
+    image_info i_info;
+    int32 image_cookie = 0;
+    int ret = 0;
+	while(get_next_image_info(team_id, &image_cookie, &i_info) == B_OK)
```

Comment:
Nit: Code style in this particular function seems to be very different from everything else, even in this file. Generally, the project uses two spaces instead of four, a space between `while` or `if` and the opening paren, and `{` on the same line as the condition. Also, what's going on with the indentation of this particular line?